app/Main.hs

```diff
@@ -351,6 +351,19 @@ runCommand options env =
     -- replaces the current process with `command`. It does not return.
     executeFile command searchPath args env'
 
+
+-- | Add Vault authentication token to a request, if set in the options. If
+-- not, the request is returned unmodified.
+addVaultToken :: Options Validated Completed -> Request -> Request
+addVaultToken options request =
+  let
+    optToken = oVaultToken options
+  in
+    case optToken of
```

Comment:
I don’t mind this `let`, but in case you did not realize, you can also just write

```haskell
case oVaultToken options of
  ...
```

Reply:
Thanks! I think I like the `let` expression here, so I think I'll leave it as-is.